Library/Tracking/KSRAnalytics.swift

```diff
@@ -1239,6 +1239,11 @@ public final class KSRAnalytics {
     properties: [String: Any] = [:],
     refTag: String? = nil
   ) {
+    if featureConsentManagementDialogEnabled() {
+      guard AppEnvironment.current.appTrackingTransparency.authorizationStatus() == .authorized
+      else { return }
+    }
+
```

Comment:
can be cleaner with 
```
guard featureConsentManagementDialogEnabled(), AppEnvironment.current.appTrackingTransparency.authorizationStatus() == .authorized else { return }
```

Reply:
So given we're requiring IDFA here, would KS want to get IDFA on every call? If that's the case then we can add it to `userProperties` which is sent up on every call.

Reply:
I'm not sure I understand. We aren't requiring IDFA here

Reply:
Do you think this as simple as adding 
`props["advertising_identifer"] = AppEnvironment.current.appTrackingTransparency.advertisingIdentifier`
to the userProperties method in KSRAnalytics? Or would it require updating the user model with a new property and potentially add too much to this current PR?